### ISSUE TITLE
Support for non-standard mono installation path

### ISSUE THREAD

**Issue by puremourning** [OmniSharp/omnisharp-server#179]
Due to various version compatibilities and standard server builds I was not able to get a system `mono` installation on RHEL 6 which can support OmniSharpServer.

So, following the instructions on mono's web page I installed mono in a custom location (e.g. $HOME/opt/mono). However, OmniSharpServer was unable to start. This is because OmniSharpServer contains [hard-coded paths](https://github.com/OmniSharp/omnisharp-server/blob/master/OmniSharp/Solution/AssemblySearch.cs) to search.

I have a local modification to `Project::FindAssembly` this which checks for the presence of a hierarchy of environment variables (`OMNISHARP_LIBRARY_PATH`, `LD_LIBRARY_PATH`,`DYLIB_FALLBACK_LIBRARY_PATH`) and appends a series of suffixes (mono/4, mono/3.5, etc.) to search for the requested assembly. This allowed me to use mono in a non-standard path by following the instructions on mono's website (they suggest setting `LD_LIBRARY_PATH`).

Has this come up before? Would you be interested in this as a pull request?

**Comment by nosami** [comment]
Yes, it has come up before, and yes, I would be interested in a pull request!!

Bear in mind though that this project is almost obsolete (to be replaced with https://github.com/OmniSharp/omnisharp-roslyn). I think your problem isn't an issue there.
